modules/standard/IO.chpl

```diff
@@ -1971,7 +1961,54 @@ proc =(ref ret:channel, x:channel) {
 }
 
 pragma "no doc"
-proc channel.channel(param writing:bool, param kind:iokind, param locking:bool, f:file, out error:syserr, hints:c_int, start:int(64), end:int(64), in local_style:iostyle) {
+proc channel.init(param writing:bool, param kind:iokind, param locking:bool) {
+  this.writing = writing;
+  this.kind = kind;
+  this.locking = locking;
+  super.init();
+}
+
+pragma "no doc"
+proc channel.init(x: channel) {
+  this.writing = x.writing;
+  this.kind = x.kind;
+  this.locking = x.locking;
+  this.home = x.home;
+  this._channel_internal = x._channel_internal;
+  _readWriteThisFromLocale = x._readWriteThisFromLocale;
+  super.init();
+  on x.home {
+    qio_channel_retain(x._channel_internal);
+  }
+}
+
+//
+// Note that this is effectively the initializer that the compiler
+// would typically provide and that, by providing the next initializer
+// below, we have to write it out manually...  A good case for having
+// a means to "opt-in" to including the compiler-provided initializer?
```

Comment:
I think it makes sense for this to be an explicitly written initializer because it should be 'private'. Can you check to see if making it 'private' works here? AFAIK it should only be called from _read_one_internal / _write_one_internal.

Reply:
I thought private couldn't be applied to methods yet, am I misremembering?

Reply:
No you're right, it gives a compilation error today. But I think it's only not implemented because we've got the wrong idea about what 'private' should mean. I think it should always mean module-private and we shouldn't bother with C++ or Java style field private. But of course that's another issue.

Reply:
Nah, it's only not implemented because we haven't made an official decision yet :)